specification/cpim/resource-manager/Microsoft.AzureActiveDirectory/preview/2023-05-17-preview/externalIdentities.json

```diff
@@ -0,0 +1,1808 @@
+      ],
+      "properties": {
+        "id": {
+          "description": "Fully qualified ID for the async operation.",
+          "type": "string"
+        },
+        "name": {
+          "description": "Name of the async operation.",
+          "type": "string"
+        },
+        "status": {
+          "description": "Operation status.",
+          "type": "string"
+        },
+        "startTime": {
+          "description": "Start time of the async operation.",
+          "type": "string"
+        },
+        "endTime": {
+          "description": "End time of the async operation.",
+          "type": "string"
+        },
+        "error": {
+          "x-nullable": true,
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorDetail"
+        }
+      }
+    },
+    "ProvisioningState": {
+      "title": "Provisioning State",
+      "type": "string",
+      "readOnly": true,
+      "enum": [
+        "Succeeded",
+        "Failed",
+        "Canceled",
+        "Provisioning",
+        "Deleting",
+        "Deleted"
+      ],
+      "x-ms-enum": {
+        "name": "ProvisioningState",
+        "modelAsString": true
+      }
+    },
+    "CIAMTenantResource": {
+      "type": "object",
+      "description": "The Azure AD for customers resource.",
+      "required": [
+        "location",
+        "sku",
+        "properties"
+      ],
+      "x-ms-azure-resource": true,
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "An identifier that represents the Azure AD for customers tenant resource."
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The name of the Azure AD for customers tenant resource."
+        },
+        "type": {
+          "type": "string",
+          "description": "The type of the Azure AD for customers tenant resource.",
+          "readOnly": true,
+          "enum": [
+            "Microsoft.AzureActiveDirectory/ciamDirectories"
+          ],
+          "x-ms-enum": {
+            "name": "ciamTypeValue",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "Microsoft.AzureActiveDirectory/ciamDirectories",
+                "description": "The resource type for Azure AD for customers tenant resource."
+              }
+            ]
+          }
+        },
+        "location": {
+          "type": "string",
+          "description": "The location in which the resource is hosted and data resides. Can be one of 'United States', 'Europe', 'Asia Pacific', or 'Australia'. Refer to [this documentation](https://aka.ms/ciam-data-location) for more information.",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ],
+          "x-ms-enum": {
+            "name": "ciamLocationValue",
+            "modelAsString": true,
+            "values": [
```

Comment:
this need not be am enum. Also the values need to match the Azure region list and not the geographies

Reply:
We don't use Azure Regions and instead use geographies. We already do this for the B2C Resource. Is there a reason to not do that?

Reply:
If we change those values that will break everything on our end.

Reply:
Ah! In that case your previous definition was better :) Sorry about that!! I am signing off now bit pl feel free to revert to the previous definition.

Reply:
Oh so I can just copy what B2C does for this field? Where location was just a string

Reply:
oh wait I understand now. Ill restore my custom location property

Reply:
yeah Ill have to revert using the TrackedResource type as well since it won't let me overwrite the location property in SDK generation